### DISCUSSION TITLE
suitable for use with docker swarm?

### DISCUSSION THREAD

**Discussion by metaljo5001** [aws/credentials-fetcher#40] [Q&A]
Hello,

A bit of setup first, we are currently using a fully windows docker swarm and are considering migrating to a hybrid with some windows and some linux nodes. Would this project be suitable to migrate workloads from windows nodes to linux nodes (using GMSA) on a swarm? If we do and would need to add contributions to this project, would they be considered/accepted?

**Comment by smhmhmd** [comment]
Yes, contributions are welcome, please reach out to us samiull at amazon.com or saakla at amazon.com

**Reply by metaljo5001** [reply]
alright! what do you think about using this project for docker swarm? do you think that it would work as is or would it require something more?

**Reply by smhmhmd** [reply]
As long as docker containers can share a bind mount in swarm-mode, it should work.
Are there any restrictions on docker when running in swarm-mode ?

**Reply by metaljo5001** [reply]
From what I can find, using a bind mount in swarm-mode is indeed possible. thank you!

**Reply by smhmhmd** [reply]
Sounds good, please let us know how it went for you.

**Reply by saikiranakula-amzn** [reply]
for the swarm mode, yes bind mounts are possible. But the only thing you might need to look at is, which kerberos tickets corresponds to which container. Current apis for credentials-fetcher provides you output of created kerberos ticket paths based on the credentialspec input, it seems the use-case provided is feasible. Please reach out to either of us if you need any additional information needed for contribution. Thank you :)
